course5/homework6/mystr.cc

```diff
@@ -0,0 +1,104 @@
+#include "mystr.h"
+#include <memory>
+#include <cmath> 
+#include <cstring>
+
+using namespace mystring;
+
+MyString::MyString()
+{
+	mStr = std::make_unique<char[]>(1); //length 1 for zero terminator
+	mStr[0] = '\0';
+}
+
+// Function to create a new unique pointer, memset it and initialize it with a 
+// new string. 
+// Repeatedly needed for copy/move assignment and construction
+void MyString::Reset_mStr(int len, const char* str)
+{
+	mStr = std::make_unique<char[]>(len+1); // add 1 for \0 termination
+	std::memset(mStr.get(), 0, len + 1);
```

Comment:
Why zero all the bytes, if you are immediately override them with another content?